src/platform/packages/shared/kbn-elastic-agent-utils/src/agent_sdk_name_and_language.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { isElasticAgentName, isOpenTelemetryAgentName } from './agent_guards';
+
+interface SdkNameAndLanguage {
+  sdkName?: 'apm' | 'edot' | 'otel_other';
+  language?: string;
+}
+
+const LANGUAGE_INDEX = 1;
+
+export const getSdkNameAndLanguage = (agentName: string): SdkNameAndLanguage => {
+  if (isElasticAgentName(agentName)) {
+    return { sdkName: 'apm', language: agentName };
+  }
+  const agentNameParts = agentName.split('/');
+
+  if (isOpenTelemetryAgentName(agentName)) {
+    if (agentNameParts[agentNameParts.length - 1] === 'elastic') {
+      return { sdkName: 'edot', language: agentNameParts[LANGUAGE_INDEX] };
+    }
+    return {
+      sdkName: 'otel_other',
+      language: agentNameParts[LANGUAGE_INDEX],
```

Comment:
Because of how OTel agent names are [defined here](https://github.com/elastic/kibana/blob/75f9c6113d7815f542f97ad37dcec7ffe6263f23/src/platform/packages/shared/kbn-elastic-agent-utils/src/agent_names.ts#L46), the function [isOpenTelemetryAgentName](https://github.com/elastic/kibana/blob/75f9c6113d7815f542f97ad37dcec7ffe6263f23/src/platform/packages/shared/kbn-elastic-agent-utils/src/agent_guards.ts#L53) will return `true` also for values such as `otlp` or `opentelemetry`.

Wouldn't this line then throw an error because of accessing an invalid index in `agentNameParts` ?

Reply:
If we have just `otlp` or `opentelemetry` with missing language we don't have a dashboard because the language is not set - so this will be `undefined` and when we do the [next check](https://github.com/elastic/kibana/pull/211822/files/cd8bde4d760728c898930523543ac5dd641f0d08#diff-884c2ba71862b74ea3ba49388b03e2319289df19204ed4edf9130b56f71aa7fdR29) we will also have the filename set as `undefined` so we want go into the logic where we are showing the dashboard - we will show the empty case instead. - I added a [test](https://github.com/elastic/kibana/pull/211822/files/cd8bde4d760728c898930523543ac5dd641f0d08#diff-3e8c9691b866b603396c811183c3fdb7153cecc8d61b44867c545d09ff79593bR39-R41) covering such case (along with other scenarios)